### ISSUE TITLE
2.0.x don't work

### ISSUE THREAD

**Issue by Jeffsyl01** [arangodb/spring-data#46]
I try to use Spring Data ArangoDB 2.0.0 in spring-data-demo-master.
There is a ParseException when running CrudRunner .

Exception detail:

```
Caused by: com.arangodb.ArangoDBException: java.text.ParseException: Unparseable date: "Ned"
	at com.arangodb.springframework.core.convert.TimeStringConverters.parse(TimeStringConverters.java:67) ~[arangodb-spring-data-2.0.0.jar:na]
	at com.arangodb.springframework.core.convert.TimeStringConverters.access$000(TimeStringConverters.java:44) ~[arangodb-spring-data-2.0.0.jar:na]
	at com.arangodb.springframework.core.convert.TimeStringConverters$StringToInstantConverter.convert(TimeStringConverters.java:105) ~[arangodb-spring-data-2.0.0.jar:na]
	at com.arangodb.springframework.core.convert.TimeStringConverters$StringToInstantConverter.convert(TimeStringConverters.java:100) ~[arangodb-spring-data-2.0.0.jar:na]
	at org.springframework.core.convert.support.GenericConversionService$ConverterAdapter.convert(GenericConversionService.java:385) ~[spring-core-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.core.convert.support.ConversionUtils.invokeConverter(ConversionUtils.java:40) ~[spring-core-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	... 40 common frames omitted
Caused by: java.text.ParseException: Unparseable date: "Ned"
	at java.text.DateFormat.parse(DateFormat.java:366) ~[na:1.8.0_161]
	at com.arangodb.velocypack.internal.util.DateUtil.parse(DateUtil.java:60) ~[velocypack-1.0.14.jar:na]
	at com.arangodb.springframework.core.convert.TimeStringConverters.parse(TimeStringConverters.java:65) ~[arangodb-spring-data-2.0.0.jar:na]
	... 45 common frames omitted
```

pom :
		
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>2.0.2.RELEASE</version>
	</parent>
	<properties>
		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
		<java.version>1.8</java.version>
	</properties>
	<dependencies>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter</artifactId>
		</dependency>
		<dependency>
			<groupId>com.arangodb</groupId>
			<artifactId>arangodb-spring-data</artifactId>
			<version>2.0.0</version>
		</dependency>
	</dependencies>

**Comment by christian-lechner** [comment]
Can you use the `2.0.0.RELEASE` of `spring-boot-starter-parent`? With `2.0.0` it seems to work. Maybe they changed the behavior of converter handling. I will have a look at this.

**Comment by mpv1989** [comment]
Yes. It looks like there is now the need of annotating `Converter`s with `@ReadingConverter` otherwise they are used for writing. I committed a [fix](https://github.com/arangodb/spring-data/commit/b649dbece229e17dd39ad303e57746078fabee2b)

**Comment by mpv1989** [comment]
I also upgraded our spring demo. It now uses arangodb-spring-data `2.1.3` and spring-boot-starter-parent `2.0.1.RELEASE`.

**Comment by christian-lechner** [comment]
The changelog of the `2.0.7.RELEASE` of Spring Data Commons says:
`* DATACMNS-1294 - Consider types in java.time to be simple.`

They did the following in `SimpleTypeHolder#isSimpleType`:
```java
if (typeName.startsWith("java.lang") || type.getName().startsWith("java.time")) {
	return true;
}
```

Instead of adding the `java.time` classes to their default type array (the registration of the default types can be prevented by passing `false` to the constructor), they added the check for `java.time` directly to the `isSimpleType` method.

The hardcoding of this check is not optimal, as this means that every Spring Data store needs to support `java.time` classes out of the box. But since we are on Java 8, it may be ok.

The writing converters in `TimeStringConverters` are thus not used anymore. The workaround with `@ReadingConverter` does solve this issue.

@mpv1989 as `java.time` classes are now considered to be simple types, should we move the reading converters from `TimeStringConverters` to `ArangoSimpleTypeConverters` and remove the writing converters as they have no effect anymore?

**Comment by mpv1989** [comment]
@christian-lechner the writing converters in `TimeStringConverters` are still used. I don't know whether we should remove them. They behave different for `LocalDate` and `LocaeDateTime` like in Spring Data core where they are converted in format `yyyy-MM-dd'T'HH:mm:ss.SSS'Z'`. Our converters behave like described in #36

**Comment by christian-lechner** [comment]
You are right, they are still used. What I meant was: they should have no effect, since they are now simple types they should be passed to the Java driver as they are.

But Spring registers the `org.springframework.data.convert.Jsr310Converters` by default which does a different conversion as you wrote. So we need our writing converters ;)

**Comment by mpv1989** [comment]
@christian-lechner 

> the ADDITIONAL_DESERIALIZABLE_TYPES from ArangoExtCursorIterator can now also be removed.

I would leave them in for backward compatibility.

> But not all types are supported by the VPackJdk8Module. Maybe we should add them, at least the "important" ones like Duration and Period (LocalTime, OffsetTime)?

Good point.

Speaking of `ArangoCursor` and Spring Data `2.0.7-RELEASE`, there is a [commit](https://github.com/spring-projects/spring-data-commons/commit/5986b4b7afc84105bdd45c26e9a59c708b975506) which removes the possibility to use `ArangoCursor` as a return type for repository methods. :-(

**Comment by christian-lechner** [comment]
> I would leave them in for backward compatibility.

Or add them to `ArangoSimpleTypes`.

> Speaking of ArangoCursor and Spring Data 2.0.7-RELEASE, there is a commit which removes the possibility to use ArangoCursor as a return type for repository methods. :-(

I see. The generic type argument of `ArangoCursor` is not unwrapped anymore. That's not that much of a problem, since we can override the `QueryMethod#getReturnedObjectType()` method in `ArangoQueryMethod`:

```java
private final TypeInformation<?> returnType;

public ArangoQueryMethod(final Method method, final RepositoryMetadata metadata, final ProjectionFactory factory) {
	super(method, metadata, factory);
	this.method = method;
	this.returnType = ClassTypeInformation.fromReturnTypeOf(method);
}
// ...
@Override
public Class<?> getReturnedObjectType() {
	if (ArangoCursor.class.equals(method.getReturnType())) {
		return returnType.getRequiredComponentType().getType();
	}
	return super.getReturnedObjectType();
}
```

I think this should do the trick?

**Comment by christian-lechner** [comment]
Ok this leads to a NPE. As the super constructor creates a `ResultProcessor` which uses the `getReturnedObjectType()` method. But the member var `method` of `ArangoQueryMethod` is not yet set.

**Comment by christian-lechner** [comment]
The only thing we can do for now is in `ArangoQueryMethod`:
```java
TypeInformation<?> getReturnType() {
	return returnType;
}
```

And in `AbstractArangoQuery#getResultClass()`:
```java
if (ArangoCursor.class.isAssignableFrom(method.getReturnType().getType())) {
	return method.getReturnType().getRequiredComponentType().getType();
}
```

**Comment by mpv1989** [comment]
Yes, I also see no other (not dirty) way.

**Comment by christian-lechner** [comment]
@mpv1989 I created an issue for this: https://jira.spring.io/browse/DATACMNS-1323

Let's see what they say.

**Comment by christian-lechner** [comment]
@mpv1989 we should continue in #47.

**Comment by mpv1989** [comment]
@Jeffsyl01 I just released version `2.1.4` which is compatible with Spring `5.0.2.RELEASE` and Spring Data `2.0.7.RELEASE`
